External/FEXCore/Source/Interface/Context/Context.h

```diff
@@ -269,9 +269,22 @@ namespace FEXCore::Context {
 
     uint8_t GetGPRSize() const { return Config.Is64BitMode ? 8 : 4; }
 
+    std::shared_mutex MemoryEntryMutex{};
+    struct MemoryEntry {
+      uint64_t Length;
+      bool Writable;
+    };
+
     void AddNamedRegion(uintptr_t Base, uintptr_t Size, uintptr_t Offset, const std::string &filename);
     void RemoveNamedRegion(uintptr_t Base, uintptr_t Size);
 
+    std::map<uint64_t, MemoryEntry> MemoryMaps;
```

Comment:
Could you add a comment documenting what the map key is? (Also should include that the key is a multiple of the page-size)